pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java

```diff
@@ -627,6 +627,8 @@ protected void validateFunctionConfigs(FunctionConfig functionConfig) {
         protected Integer instanceIdOffset = 0;
         @Parameter(names = "--runtime", description = "either THREAD or PROCESS. Only applies for Java functions")
         protected String runtime;
+        @Parameter(names = "--environment-based-secrets-provider", description = "Should we use environment based secret provider")
```

Comment:
I think there is already too arguments for the CLI.  Lets try to create something more generic.  Instead of creating an argument for a specific provider, can we create an argument to allow users to specify the secret provider type?  In the future, if we add more providers, I don't want to keep add one-of arguments.  For example, can we create a enum?

Enum SecretsProviderType {
ENV,
DEFAULT,
...
}

based on the enum determine

Reply:
Actually we should just let the user to specify class of the secrets configurator to use.  That is simplest and generic.

--secrets-provider-configurator-classname

Reply:
Changed